core/src/test/java/org/apache/iceberg/TestMetadataTableScans.java

```diff
@@ -805,6 +807,47 @@ public void testPartitionSpecEvolutionRemoval() {
     }
   }
 
+  @Test
+  public void testPartitionSpecEvolutionToUnpartitioned() throws IOException {
```

Comment:
Actually we have a test specifically for this: TestMetadataTableScansWithPartitionEvolution, for organization purpose?  Could we move it there? 

Also another question, does this test the case?  I thought it was only failing on V2, because in V1 the partition transform remains as void on the latest spec, when removed.

Reply:
I moved it now. 

> Also another question, does this test the case? I thought it was only failing on V2, because I had seen on V1 the partition transform remains as void on the latest spec, when removed.

Yes, the testcase fails only for v2 without the PR changes as V1 will have void transform to satisfy the check. But I thought it can run for both.

BTW, now I removed the scan code in the testcase because my intention was to just scan spec_id column and see it is filled instead of null. But I don't know how to achieve it (In sql or spark code I know. But in this table scan I am not sure as instead of rows it returns content files)

Reply:
ok. Added it back.